app/controllers/api/v1/orders_controller.rb

```diff
@@ -69,12 +69,7 @@ def show
       end
 
       def transition
-        transition_event = params['transition'].to_sym
-        cancellation_reason = params["cancellation_reason"]
-        if @order.state_events.include?(transition_event)
-          @order.fire_state_event(transition_event)
-          @order.update(cancellation_reason: cancellation_reason) if cancellation_reason.presence
-        end
+        update_transition_and_cancel_reason
```

Comment:
I would rather not move this code into a separate `update_transition_and_cancel_reason`. It's now only 2 lines and doesn't warrant it's own method unless it's being reused elsewhere but I don't see that. IT makes it hard to follow the code path if having to jump around too much, esp with many short methods.